### PR TITLE
Mark generic top level prefix proxy deprecated

### PR DESCRIPTION
**Pull request by feihujiang**
this ref #15830, #18009

### CODE REVIEW

File: pkg/apiserver/api_installer.go [hunk 1]
```diff
@@ -323,7 +323,11 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		actions = appendIf(actions, action{"PATCH", itemPath, nameParams, namer}, isPatcher)
 		actions = appendIf(actions, action{"DELETE", itemPath, nameParams, namer}, isDeleter)
 		actions = appendIf(actions, action{"WATCH", "watch/" + itemPath, nameParams, namer}, isWatcher)
+		// We add "proxy" subresource to remove the need for the generic top level prefix proxy.
+		// The generic top level prefix proxy will be deprecated for v1.2.
```
Comment (lavalamp) [inline pkg/apiserver/api_installer.go]: "The generic top level prefix proxy is deprecated in v1.2, and will be removed in 1.3, or 1.4 at the latest."

Comment (feihujiang) [inline pkg/apiserver/api_installer.go]: Done.

**Comment by feihujiang** [comment]
@lavalamp Addressed your comment, PTAL.

**Comment by lavalamp** [comment]
LGTM, thanks!

**Pull request merged**